subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/image_info.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2018-2021 mcumgr authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_IMG_MGMT_INFO_
+#define H_IMG_MGMT_INFO_
+
+#include <inttypes.h>
+#include "image.h"
+#include "mgmt/mgmt.h"
+#include <zcbor_common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* 255.255.65535.4294967295\0 */
+#define IMG_MGMT_VER_MAX_STR_LEN	(sizeof("255.255.65535.4294967295"))
+
+/*
+ * @brief Read info of an image give the slot number
+ *
+ * @param image_slot	 Image slot to read info from
+ * @param image_version  Image version to be filled up
+ * @param hash		   Ptr to the read image hash
+ * @param flags		  Ptr to flags filled up from the image
+ */
+int img_mgmt_read_info(int image_slot, struct image_version *ver, uint8_t *hash, uint32_t *flags);
```

Comment:
These should probably go in `zephyr/include` so that documentation can be generated

Reply:
I have considered adding this on top of the 'flatten' PR, but that one has already been delayed too long, so decided to go separately with this.
These sets of functions have two problems:
 1) These are internal mcumgr utility functions, that may be used outside and have been exported via header for system specific ports to use, something that is no longer required;
 2) MCUmgr shouldn't even provide these; these should be provided by DFU subsystem or MCUboot library and MCUmgr should be a client to them.

The naming `image_state` and `image_info` has been selected for that reason: these are not `img_mgmt` functions at all.
Registration functions are, callback are but these functions are img_mgmt independent.
Image info functions work on image directly and state functions are not  `img_mgmt_state` functions,, if you look at "image list" that is image state command, these set image state as defined by MCUboot, and work on image directly - they do not require mcumgr at all.

API defined by `image_info.h` and `image_state.h` should be moved to DFU subsystem or be part of MCUboot bootutil API.